// vue3-google-map/dist/index.mjs
(function(){var e;"use strict";try{if(typeof document<"u"){var a=document.createElement("style");a.nonce=(e=document.head.querySelector("meta[property=csp-nonce]"))==null?void 0:e.content,a.appendChild(document.createTextNode(".mapdiv[data-v-4e398d5a]{width:100%;height:100%}.advanced-marker-wrapper{display:none}.mapdiv .advanced-marker-wrapper{display:inline-block}.custom-control-wrapper[data-v-ab9120cd]{display:none}.mapdiv .custom-control-wrapper[data-v-ab9120cd]{display:inline-block}.info-window-wrapper[data-v-4d4142cf]{display:none}.mapdiv .info-window-wrapper[data-v-4d4142cf]{display:inline-block}.custom-marker-wrapper[data-v-574a274a]{display:none}.mapdiv .custom-marker-wrapper[data-v-574a274a]{display:inline-block}")),document.head.appendChild(a)}}catch(d){console.error("vite-plugin-css-injected-by-js",d)}})();
var ht = Object.defineProperty;
var gt = (t, e, r) => e in t ? ht(t, e, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[e] = r;
var ae = (t, e, r) => (gt(t, typeof e != "symbol" ? e + "" : e, r), r);
import { defineComponent as T, ref as _, provide as U, watch as S, onMounted as nt, onBeforeUnmount as B, markRaw as A, toRef as R, openBlock as F, createElementBlock as V, createElementVNode as pe, renderSlot as z, normalizeProps as vt, guardReactiveProps as yt, computed as W, Comment as ke, inject as b, Fragment as _t, mergeProps as we, createCommentVNode as Ce } from "vue";
const D = Symbol("map"), Z = Symbol("api"), be = Symbol("marker"), Oe = Symbol("markerCluster"), fe = Symbol("CustomMarker"), st = Symbol("mapTilesLoaded"), H = [
  "click",
  "dblclick",
  "drag",
  "dragend",
  "dragstart",
  "mousedown",
  "mousemove",
  "mouseout",
  "mouseover",
  "mouseup",
  "rightclick"
];
function Et(t, e) {
  t.src = e;
}
const kt = (t) => {
  var e, r, n, s = "The Google Maps JavaScript API", o = "google", i = "importLibrary", l = "__ib__", a = document, c = window, p = c[o] || (c[o] = {}), u = p.maps || (p.maps = {}), f = /* @__PURE__ */ new Set(), m = new URLSearchParams(), h = () => e || (e = new Promise(async (d, g) => {
    var v;
    await (r = a.createElement("script")), m.set("libraries", [...f] + "");
    for (n in t)
      m.set(n.replace(/[A-Z]/g, (y) => "_" + y[0].toLowerCase()), t[n]);
    m.set("callback", o + ".maps." + l), Et(r, "https://maps.googleapis.com/maps/api/js?" + m), u[l] = d, r.onerror = () => e = g(Error(s + " could not load.")), r.nonce = ((v = a.querySelector("script[nonce]")) == null ? void 0 : v.nonce) || "", a.head.append(r);
  }));
  u[i] ? console.warn(s + " only loads once. Ignoring:", t) : u[i] = (d, ...g) => f.add(d) && h().then(() => u[i](d, ...g));
}, wt = (t) => `The setOptions() function should only be called once. The options passed to the additional call (${JSON.stringify(t)}) will be ignored.`, Ct = (t) => `The google.maps.importLibrary() function is already defined, and @googlemaps/js-api-loader will use the existing function instead of overwriting it. The options passed to setOptions (${JSON.stringify(t)}) will be ignored.`, bt = "No options were set before calling importLibrary. Make sure to configure the loader using setOptions().", Ot = "There already is a script loading the Google Maps JavaScript API, and no google.maps.importLibrary function is defined. @googlemaps/js-api-loader will proceed to bootstrap the API with the specified options, but the existing script might cause problems using the API. Make sure to remove the script loading the API.", Me = process.env.NODE_ENV !== "production" ? (t) => {
  console.warn(`[@googlemaps/js-api-loader] ${t}`);
} : () => {
}, Mt = process.env.NODE_ENV !== "production" ? (t) => {
  console.info(`[@googlemaps/js-api-loader] ${t}`);
} : () => {
};
let ye = !1;
function St(t) {
  if (ye) {
    Me(wt(t));
    return;
  }
  Pt(t), ye = !0;
}
async function Tt(t) {
  var e, r;
  if (ye || Me(bt), !((r = (e = window == null ? void 0 : window.google) == null ? void 0 : e.maps) != null && r.importLibrary))
    throw new Error("google.maps.importLibrary is not installed.");
  return await google.maps.importLibrary(t);
}
function Pt(t) {
  var r, n;
  const e = !!((n = (r = window.google) == null ? void 0 : r.maps) != null && n.importLibrary);
  e ? Mt(Ct(t)) : process.env.NODE_ENV !== "production" && document.querySelector('script[src*="maps.googleapis.com/maps/api/js"]') && Me(Ot), e || kt(t);
}
function At(t) {
  return class extends t.OverlayView {
    constructor(n) {
      super();
      ae(this, "element");
      ae(this, "opts");
      const { element: s, ...o } = n;
      this.element = s, this.opts = o, this.opts.map && this.setMap(this.opts.map);
    }
    getPosition() {
      return this.opts.position ? this.opts.position instanceof t.LatLng ? this.opts.position : new t.LatLng(this.opts.position) : null;
    }
    getVisible() {
      if (!this.element)
        return !1;
      const n = this.element;
      return n.style.display !== "none" && n.style.visibility !== "hidden" && (n.style.opacity === "" || Number(n.style.opacity) > 0.01);
    }
    onAdd() {
      if (!this.element)
        return;
      const n = this.getPanes();
      n && n.overlayMouseTarget.appendChild(this.element);
    }
    draw() {
      if (!this.element)
        return;
      const n = this.getProjection(), s = n == null ? void 0 : n.fromLatLngToDivPixel(this.getPosition());
      if (s) {
        this.element.style.position = "absolute";
        let o, i;
        switch (this.opts.anchorPoint) {
          case "TOP_CENTER":
            o = "-50%", i = "-100%";
            break;
          case "BOTTOM_CENTER":
            o = "-50%", i = "0";
            break;
          case "LEFT_CENTER":
            o = "-100%", i = "-50%";
            break;
          case "RIGHT_CENTER":
            o = "0", i = "-50%";
            break;
          case "TOP_LEFT":
            o = "-100%", i = "-100%";
            break;
          case "TOP_RIGHT":
            o = "0", i = "-100%";
            break;
          case "BOTTOM_LEFT":
            o = "-100%", i = "0";
            break;
          case "BOTTOM_RIGHT":
            o = "0", i = "0";
            break;
          default:
            o = "-50%", i = "-50%";
        }
        const l = s.x + (this.opts.offsetX || 0) + "px", a = s.y + (this.opts.offsetY || 0) + "px";
        this.element.style.transform = `translateX(${o}) translateX(${l}) translateY(${i}) translateY(${a})`, this.opts.zIndex && (this.element.style.zIndex = this.opts.zIndex.toString());
      }
    }
    onRemove() {
      this.element && this.element.remove();
    }
    setOptions(n) {
      const { element: s, ...o } = n;
      this.element = s, this.opts = o, this.draw();
    }
  };
}
let Ae = !1;
const qe = [
  "bounds_changed",
  "center_changed",
  "click",
  "contextmenu",
  "dblclick",
  "drag",
  "dragend",
  "dragstart",
  "heading_changed",
  "idle",
  "isfractionalzoomenabled_changed",
  "mapcapabilities_changed",
  "maptypeid_changed",
  "mousemove",
  "mouseout",
  "mouseover",
  "projection_changed",
  "renderingtype_changed",
  "rightclick",
  "tilesloaded",
  "tilt_changed",
  "zoom_changed"
], qt = T({
  props: {
    apiPromise: {
      type: Promise
    },
    apiKey: {
      type: String,
      default: ""
    },
    version: {
      type: String,
      default: "weekly"
    },
    libraries: {
      type: Array,
      default: () => ["places", "marker"]
    },
    region: {
      type: String,
      required: !1
    },
    language: {
      type: String,
      required: !1
    },
    backgroundColor: {
      type: String,
      required: !1
    },
    center: {
      type: Object,
      default: () => ({ lat: 0, lng: 0 })
    },
    clickableIcons: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    colorScheme: {
      type: String,
      required: !1
    },
    controlSize: {
      type: Number,
      required: !1
    },
    disableDefaultUi: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    disableDoubleClickZoom: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    draggable: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    draggableCursor: {
      type: String,
      required: !1
    },
    draggingCursor: {
      type: String,
      required: !1
    },
    fullscreenControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    fullscreenControlPosition: {
      type: String,
      required: !1
    },
    gestureHandling: {
      type: String,
      required: !1
    },
    heading: {
      type: Number,
      required: !1
    },
    isFractionalZoomEnabled: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    keyboardShortcuts: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    mapTypeControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    mapTypeControlOptions: {
      type: Object,
      required: !1
    },
    mapTypeId: {
      type: [Number, String],
      required: !1
    },
    mapId: {
      type: String,
      required: !1
    },
    maxZoom: {
      type: Number,
      required: !1
    },
    minZoom: {
      type: Number,
      required: !1
    },
    noClear: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    panControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    panControlPosition: {
      type: String,
      required: !1
    },
    restriction: {
      type: Object,
      required: !1
    },
    rotateControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    rotateControlPosition: {
      type: String,
      required: !1
    },
    scaleControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    scaleControlStyle: {
      type: Number,
      required: !1
    },
    scrollwheel: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    streetView: {
      type: Object,
      required: !1
    },
    streetViewControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    streetViewControlPosition: {
      type: String,
      required: !1
    },
    styles: {
      type: Array,
      required: !1
    },
    tilt: {
      type: Number,
      required: !1
    },
    zoom: {
      type: Number,
      required: !1
    },
    zoomControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    zoomControlPosition: {
      type: String,
      required: !1
    },
    cameraControl: {
      type: Boolean,
      required: !1,
      default: void 0
    },
    cameraControlPosition: {
      type: String,
      required: !1
    }
  },
  emits: [...qe],
  setup(t, { emit: e }) {
    const r = _(), n = _(!1), s = _(), o = _(), i = _(!1);
    U(D, s), U(Z, o), U(st, i);
    const l = () => {
      const u = { ...t };
      Object.keys(u).forEach((d) => {
        u[d] === void 0 && delete u[d];
      });
      const m = (d) => {
        var g;
        return d ? { position: (g = o.value) == null ? void 0 : g.ControlPosition[d] } : {};
      }, h = {
        scaleControlOptions: t.scaleControlStyle ? { style: t.scaleControlStyle } : {},
        panControlOptions: m(t.panControlPosition),
        zoomControlOptions: m(t.zoomControlPosition),
        rotateControlOptions: m(t.rotateControlPosition),
        streetViewControlOptions: m(t.streetViewControlPosition),
        fullscreenControlOptions: m(t.fullscreenControlPosition),
        cameraControlOptions: m(t.cameraControlPosition),
        disableDefaultUI: t.disableDefaultUi
      };
      return { ...u, ...h };
    }, a = S(
      [o, s],
      ([u, f]) => {
        const m = u, h = f;
        m && h && (m.event.addListenerOnce(h, "tilesloaded", () => {
          i.value = !0;
        }), setTimeout(a, 0));
      },
      { immediate: !0 }
    ), c = () => {
      const { apiKey: u, region: f, version: m, language: h, libraries: d } = t;
      St({ key: u, region: f, v: m, language: h, libraries: d });
    }, p = (u) => {
      const f = u ? u.maps : google.maps;
      o.value = A(f), s.value = A(new f.Map(r.value, l()));
      const m = At(o.value);
      o.value[fe] = m, qe.forEach((d) => {
        var g;
        (g = s.value) == null || g.addListener(d, (v) => e(d, v));
      }), n.value = !0;
      const h = Object.keys(t).filter(
        (d) => !["apiPromise", "apiKey", "version", "libraries", "region", "language", "center", "zoom"].includes(d)
      ).map((d) => R(t, d));
      S(
        [() => t.center, () => t.zoom, ...h],
        ([d, g], [v, y]) => {
          var w, M, L;
          const { center: k, zoom: P, ...O } = l();
          (w = s.value) == null || w.setOptions(O), g !== void 0 && g !== y && ((M = s.value) == null || M.setZoom(g));
          const E = !v || d.lng !== v.lng || d.lat !== v.lat;
          d && E && ((L = s.value) == null || L.panTo(d));
        }
      );
    };
    return nt(() => {
      if (t.apiPromise && t.apiPromise instanceof Promise)
        t.apiPromise.then(p);
      else {
        Ae || (c(), Ae = !0);
        const u = t.libraries && t.libraries.length > 0 ? t.libraries : ["maps", "marker"];
        Promise.all(u.map((f) => Tt(f))).then(() => p());
      }
    }), B(() => {
      var u;
      i.value = !1, s.value && ((u = o.value) == null || u.event.clearInstanceListeners(s.value));
    }), { mapRef: r, ready: n, map: s, api: o, mapTilesLoaded: i };
  }
});
const re = (t, e) => {
  const r = t.__vccOpts || t;
  for (const [n, s] of e)
    r[n] = s;
  return r;
}, Lt = {
  ref: "mapRef",
  class: "mapdiv"
};
function xt(t, e, r, n, s, o) {
  return F(), V("div", null, [
    pe("div", Lt, null, 512),
    z(t.$slots, "default", vt(yt({ ready: t.ready, map: t.map, api: t.api, mapTilesLoaded: t.mapTilesLoaded })), void 0, !0)
  ]);
}
const Sn = /* @__PURE__ */ re(qt, [["render", xt], ["__scopeId", "data-v-4e398d5a"]]);
function jt(t) {
  return t && t.__esModule && Object.prototype.hasOwnProperty.call(t, "default") ? t.default : t;
}
var It = function t(e, r) {
  if (e === r)
    return !0;
  if (e && r && typeof e == "object" && typeof r == "object") {
    if (e.constructor !== r.constructor)
      return !1;
    var n, s, o;
    if (Array.isArray(e)) {
      if (n = e.length, n != r.length)
        return !1;
      for (s = n; s-- !== 0; )
        if (!t(e[s], r[s]))
          return !1;
      return !0;
    }
    if (e.constructor === RegExp)
      return e.source === r.source && e.flags === r.flags;
    if (e.valueOf !== Object.prototype.valueOf)
      return e.valueOf() === r.valueOf();
    if (e.toString !== Object.prototype.toString)
      return e.toString() === r.toString();
    if (o = Object.keys(e), n = o.length, n !== Object.keys(r).length)
      return !1;
    for (s = n; s-- !== 0; )
      if (!Object.prototype.hasOwnProperty.call(r, o[s]))
        return !1;
    for (s = n; s-- !== 0; ) {
      var i = o[s];
      if (!t(e[i], r[i]))
        return !1;
    }
    return !0;
  }
  return e !== e && r !== r;
};
const ee = /* @__PURE__ */ jt(It), Le = "click", ot = "gmp-click", xe = ["drag", "dragend", "dragstart", ot], Rt = T({
  name: "AdvancedMarker",
  props: {
    options: {
      type: Object,
      required: !0
    },
    pinOptions: {
      type: Object,
      required: !1
    }
  },
  emits: [...xe, Le],
  setup(t, { emit: e, expose: r, slots: n }) {
    const s = _(), o = W(() => {
      var m;
      return (m = n.content) == null ? void 0 : m.call(n).some((h) => h.type !== ke);
    }), i = R(t, "options"), l = R(t, "pinOptions"), a = _(), c = b(D, _()), p = b(Z, _()), u = b(Oe, _()), f = W(
      () => !!(u.value && p.value && a.value instanceof google.maps.marker.AdvancedMarkerElement)
    );
    return S(
      [c, i, l, s],
      async (m, [h, d, g, v]) => {
        var w, M, L;
        const y = !ee(i.value, d) || !ee(l.value, g), k = s.value !== v, P = y || k || c.value !== h;
        if (!c.value || !p.value || !P || o.value && !s.value)
          return;
        const { AdvancedMarkerElement: O, PinElement: E } = p.value.marker;
        if (a.value) {
          const { map: ne, content: se, ...oe } = i.value;
          Object.assign(a.value, {
            content: o.value ? s.value : l.value ? new E(l.value) : se,
            ...oe
          }), f.value && ((w = u.value) == null || w.removeMarker(a.value), (M = u.value) == null || M.addMarker(a.value));
        } else
          o.value ? i.value.content = s.value : l.value && (i.value.content = new E(l.value)), a.value = A(new O(i.value)), f.value ? (L = u.value) == null || L.addMarker(a.value) : a.value.map = c.value, xe.forEach((ne) => {
            var se;
            (se = a.value) == null || se.addListener(ne, (oe) => {
              e(ne, oe), ne === ot && e(Le, oe);
            });
          });
      },
      {
        immediate: !0,
        flush: "post"
        // Ensure DOM updates happen before this watcher runs
      }
    ), B(() => {
      var m, h;
      a.value && ((m = p.value) == null || m.event.clearInstanceListeners(a.value), f.value ? (h = u.value) == null || h.removeMarker(a.value) : a.value.map = null);
    }), U(be, a), r({ marker: a }), { hasCustomSlotContent: o, markerRef: s };
  }
});
const Nt = {
  key: 0,
  class: "advanced-marker-wrapper"
};
function $t(t, e, r, n, s, o) {
  return F(), V(_t, null, [
    t.hasCustomSlotContent ? (F(), V("div", Nt, [
      pe("div", we({ ref: "markerRef" }, t.$attrs), [
        z(t.$slots, "content")
      ], 16)
    ])) : Ce("", !0),
    z(t.$slots, "default")
  ], 64);
}
const Tn = /* @__PURE__ */ re(Rt, [["render", $t]]), je = (t) => t === "Marker", Ie = (t) => t === fe, K = (t, e, r, n) => {
  const s = _(), o = b(D, _()), i = b(Z, _()), l = b(Oe, _()), a = W(
    () => !!(l.value && i.value && (s.value instanceof i.value.Marker || s.value instanceof i.value[fe]))
  );
  return S(
    [o, r],
    (c, [p, u]) => {
      var m, h, d;
      const f = !ee(r.value, u) || o.value !== p;
      if (!(!o.value || !i.value || !f))
        if (s.value)
          s.value.setOptions(r.value), a.value && ((m = l.value) == null || m.removeMarker(s.value), (h = l.value) == null || h.addMarker(s.value));
        else {
          if (je(t))
            s.value = A(
              new i.value[t](r.value)
            );
          else if (Ie(t)) {
            const g = r.value;
            g.element && (s.value = A(new i.value[t](g)));
          } else
            s.value = A(
              new i.value[t]({
                ...r.value,
                map: o.value
              })
            );
          s.value && (a.value ? (d = l.value) == null || d.addMarker(s.value) : (je(t) || Ie(t)) && s.value.setMap(o.value)), e.forEach((g) => {
            var v;
            (v = s.value) == null || v.addListener(g, (y) => n(g, y));
          });
        }
    },
    {
      immediate: !0,
      flush: "post"
    }
  ), B(() => {
    var c, p;
    s.value && ((c = i.value) == null || c.event.clearInstanceListeners(s.value), a.value ? (p = l.value) == null || p.removeMarker(s.value) : s.value.setMap(null));
  }), s;
}, Re = [
  "animation_changed",
  "click",
  "dblclick",
  "rightclick",
  "dragstart",
  "dragend",
  "drag",
  "mouseover",
  "mousedown",
  "mouseout",
  "mouseup",
  "draggable_changed",
  "clickable_changed",
  "contextmenu",
  "cursor_changed",
  "flat_changed",
  "zindex_changed",
  "icon_changed",
  "position_changed",
  "shape_changed",
  "title_changed",
  "visible_changed"
], Pn = T({
  name: "Marker",
  props: {
    options: {
      type: Object,
      required: !0
    }
  },
  emits: [...Re],
  setup(t, { emit: e, expose: r, slots: n }) {
    const s = R(t, "options"), o = K("Marker", Re, s, e);
    return U(be, o), r({ marker: o }), () => {
      var i;
      return (i = n.default) == null ? void 0 : i.call(n);
    };
  }
}), An = T({
  name: "Polyline",
  props: {
    options: {
      type: Object,
      required: !0
    }
  },
  emits: [...H],
  setup(t, { emit: e }) {
    const r = R(t, "options");
    return { polyline: K("Polyline", H, r, e) };
  },
  render: () => null
}), qn = T({
  name: "Polygon",
  props: {
    options: {
      type: Object,
      required: !0
    }
  },
  emits: [...H],
  setup(t, { emit: e }) {
    const r = R(t, "options");
    return { polygon: K("Polygon", H, r, e) };
  },
  render: () => null
}), Ne = [...H, "bounds_changed"], Ln = T({
  name: "Rectangle",
  props: {
    options: {
      type: Object,
      required: !0
    }
  },
  emits: [...Ne],
  setup(t, { emit: e }) {
    const r = R(t, "options");
    return { rectangle: K("Rectangle", Ne, r, e) };
  },
  render: () => null
}), $e = [...H, "center_changed", "radius_changed"], xn = T({
  name: "Circle",
  props: {
    options: {
      type: Object,
      required: !0
    }
  },
  emits: [...$e],
  setup(t, { emit: e }) {
    const r = R(t, "options");
    return { circle: K("Circle", $e, r, e) };
  },
  render: () => null
}), Bt = T({
  props: {
    position: {
      type: String,
      required: !0
    },
    index: {
      type: Number,
      default: 1
    }
  },
  emits: ["content:loaded"],
  setup(t, { emit: e }) {
    const r = _(null), n = b(D, _()), s = b(Z, _()), o = b(st, _(!1)), i = S(
      [o, s, r],
      ([c, p, u]) => {
        p && c && u && (l(t.position), e("content:loaded"), setTimeout(i, 0));
      },
      { immediate: !0 }
    ), l = (c) => {
      if (n.value && s.value && r.value) {
        const p = s.value.ControlPosition[c];
        r.value.index = t.index, n.value.controls[p].push(r.value);
      }
    }, a = (c) => {
      if (n.value && s.value) {
        let p = null;
        const u = s.value.ControlPosition[c];
        n.value.controls[u].forEach((f, m) => {
          f === r.value && (p = m);
        }), p !== null && n.value.controls[u].removeAt(p);
      }
    };
    return B(() => a(t.position)), S(
      () => t.position,
      (c, p) => {
        a(p), l(c);
      }
    ), S(
      () => t.index,
      (c) => {
        r.value && (r.value.index = c);
      }
    ), { controlRef: r };
  }
});
const Dt = {
  ref: "controlRef",
  class: "custom-control-wrapper"
};
function Zt(t, e, r, n, s, o) {
  return F(), V("div", Dt, [
    z(t.$slots, "default", {}, void 0, !0)
  ], 512);
}
const jn = /* @__PURE__ */ re(Bt, [["render", Zt], ["__scopeId", "data-v-ab9120cd"]]);
function Be(t) {
  return t instanceof google.maps.marker.AdvancedMarkerElement ? "gmp-click" : "click";
}
const De = [
  "closeclick",
  "content_changed",
  "domready",
  "position_changed",
  "visible",
  "zindex_changed"
], Gt = T({
  inheritAttrs: !1,
  props: {
    options: {
      type: Object,
      default: () => ({})
    },
    modelValue: {
      type: Boolean
    }
  },
  emits: [...De, "update:modelValue"],
  setup(t, { slots: e, emit: r, expose: n }) {
    const s = _(), o = _(), i = b(D, _()), l = b(Z, _()), a = b(be, _());
    let c, p = t.modelValue;
    const u = W(() => {
      var d;
      return (d = e.default) == null ? void 0 : d.call(e).some((g) => g.type !== ke);
    }), f = (d) => {
      p = d, r("update:modelValue", d);
    }, m = (d) => {
      s.value && (s.value.open({ map: i.value, anchor: a.value, ...d }), f(!0));
    }, h = () => {
      s.value && (s.value.close(), f(!1));
    };
    return nt(() => {
      S(
        [i, () => t.options],
        ([d, g], [v, y]) => {
          var P;
          const k = !ee(g, y) || i.value !== v;
          i.value && l.value && k && (s.value ? (s.value.setOptions({
            ...g,
            content: u.value ? o.value : g.content
          }), a.value || m()) : (s.value = A(
            new l.value.InfoWindow({
              ...g,
              content: u.value ? o.value : g.content
            })
          ), a.value && (c = a.value.addListener(Be(a.value), () => m())), (!a.value || p) && m(), De.forEach((O) => {
            var E;
            (E = s.value) == null || E.addListener(O, (w) => r(O, w));
          }), (P = s.value) == null || P.addListener("closeclick", () => f(!1))));
        },
        {
          immediate: !0,
          flush: "post"
        }
      ), S(
        a,
        (d, g) => {
          !s.value || d === g || (c && (c.remove(), c = void 0), d && (c = d.addListener(Be(d), () => m())));
        },
        {
          flush: "post"
        }
      ), S(
        () => t.modelValue,
        (d) => {
          d !== p && (d ? m() : h());
        }
      );
    }), B(() => {
      var d;
      c && c.remove(), s.value && ((d = l.value) == null || d.event.clearInstanceListeners(s.value), h());
    }), n({ infoWindow: s, open: m, close: h }), { infoWindow: s, infoWindowRef: o, hasSlotContent: u, open: m, close: h };
  }
});
const Ut = {
  key: 0,
  class: "info-window-wrapper"
};
function Ft(t, e, r, n, s, o) {
  return t.hasSlotContent ? (F(), V("div", Ut, [
    pe("div", we({ ref: "infoWindowRef" }, t.$attrs), [
      z(t.$slots, "default", {}, void 0, !0)
    ], 16)
  ])) : Ce("", !0);
}
const In = /* @__PURE__ */ re(Gt, [["render", Ft], ["__scopeId", "data-v-4d4142cf"]]);
var Vt = Object.getOwnPropertyNames, zt = Object.getOwnPropertySymbols, Wt = Object.prototype.hasOwnProperty;
function Ze(t, e) {
  return function(n, s, o) {
    return t(n, s, o) && e(n, s, o);
  };
}
function ie(t) {
  return function(r, n, s) {
    if (!r || !n || typeof r != "object" || typeof n != "object")
      return t(r, n, s);
    var o = s.cache, i = o.get(r), l = o.get(n);
    if (i && l)
      return i === n && l === r;
    o.set(r, n), o.set(n, r);
    var a = t(r, n, s);
    return o.delete(r), o.delete(n), a;
  };
}
function Ge(t) {
  return Vt(t).concat(zt(t));
}
var Ht = Object.hasOwn || function(t, e) {
  return Wt.call(t, e);
};
function G(t, e) {
  return t === e || !t && !e && t !== t && e !== e;
}
var Kt = "__v", Xt = "__o", Yt = "_owner", Ue = Object.getOwnPropertyDescriptor, Fe = Object.keys;
function Jt(t, e, r) {
  var n = t.length;
  if (e.length !== n)
    return !1;
  for (; n-- > 0; )
    if (!r.equals(t[n], e[n], n, n, t, e, r))
      return !1;
  return !0;
}
function Qt(t, e) {
  return G(t.getTime(), e.getTime());
}
function er(t, e) {
  return t.name === e.name && t.message === e.message && t.cause === e.cause && t.stack === e.stack;
}
function tr(t, e) {
  return t === e;
}
function Ve(t, e, r) {
  var n = t.size;
  if (n !== e.size)
    return !1;
  if (!n)
    return !0;
  for (var s = new Array(n), o = t.entries(), i, l, a = 0; (i = o.next()) && !i.done; ) {
    for (var c = e.entries(), p = !1, u = 0; (l = c.next()) && !l.done; ) {
      if (s[u]) {
        u++;
        continue;
      }
      var f = i.value, m = l.value;
      if (r.equals(f[0], m[0], a, u, t, e, r) && r.equals(f[1], m[1], f[0], m[0], t, e, r)) {
        p = s[u] = !0;
        break;
      }
      u++;
    }
    if (!p)
      return !1;
    a++;
  }
  return !0;
}
var rr = G;
function nr(t, e, r) {
  var n = Fe(t), s = n.length;
  if (Fe(e).length !== s)
    return !1;
  for (; s-- > 0; )
    if (!at(t, e, r, n[s]))
      return !1;
  return !0;
}
function X(t, e, r) {
  var n = Ge(t), s = n.length;
  if (Ge(e).length !== s)
    return !1;
  for (var o, i, l; s-- > 0; )
    if (o = n[s], !at(t, e, r, o) || (i = Ue(t, o), l = Ue(e, o), (i || l) && (!i || !l || i.configurable !== l.configurable || i.enumerable !== l.enumerable || i.writable !== l.writable)))
      return !1;
  return !0;
}
function sr(t, e) {
  return G(t.valueOf(), e.valueOf());
}
function or(t, e) {
  return t.source === e.source && t.flags === e.flags;
}
function ze(t, e, r) {
  var n = t.size;
  if (n !== e.size)
    return !1;
  if (!n)
    return !0;
  for (var s = new Array(n), o = t.values(), i, l; (i = o.next()) && !i.done; ) {
    for (var a = e.values(), c = !1, p = 0; (l = a.next()) && !l.done; ) {
      if (!s[p] && r.equals(i.value, l.value, i.value, l.value, t, e, r)) {
        c = s[p] = !0;
        break;
      }
      p++;
    }
    if (!c)
      return !1;
  }
  return !0;
}
function ar(t, e) {
  var r = t.length;
  if (e.length !== r)
    return !1;
  for (; r-- > 0; )
    if (t[r] !== e[r])
      return !1;
  return !0;
}
function ir(t, e) {
  return t.hostname === e.hostname && t.pathname === e.pathname && t.protocol === e.protocol && t.port === e.port && t.hash === e.hash && t.username === e.username && t.password === e.password;
}
function at(t, e, r, n) {
  return (n === Yt || n === Xt || n === Kt) && (t.$$typeof || e.$$typeof) ? !0 : Ht(e, n) && r.equals(t[n], e[n], n, n, t, e, r);
}
var lr = "[object Arguments]", ur = "[object Boolean]", cr = "[object Date]", dr = "[object Error]", pr = "[object Map]", fr = "[object Number]", mr = "[object Object]", hr = "[object RegExp]", gr = "[object Set]", vr = "[object String]", yr = "[object URL]", _r = Array.isArray, We = typeof ArrayBuffer == "function" && ArrayBuffer.isView ? ArrayBuffer.isView : null, He = Object.assign, Er = Object.prototype.toString.call.bind(Object.prototype.toString);
function kr(t) {
  var e = t.areArraysEqual, r = t.areDatesEqual, n = t.areErrorsEqual, s = t.areFunctionsEqual, o = t.areMapsEqual, i = t.areNumbersEqual, l = t.areObjectsEqual, a = t.arePrimitiveWrappersEqual, c = t.areRegExpsEqual, p = t.areSetsEqual, u = t.areTypedArraysEqual, f = t.areUrlsEqual;
  return function(h, d, g) {
    if (h === d)
      return !0;
    if (h == null || d == null)
      return !1;
    var v = typeof h;
    if (v !== typeof d)
      return !1;
    if (v !== "object")
      return v === "number" ? i(h, d, g) : v === "function" ? s(h, d, g) : !1;
    var y = h.constructor;
    if (y !== d.constructor)
      return !1;
    if (y === Object)
      return l(h, d, g);
    if (_r(h))
      return e(h, d, g);
    if (We != null && We(h))
      return u(h, d, g);
    if (y === Date)
      return r(h, d, g);
    if (y === RegExp)
      return c(h, d, g);
    if (y === Map)
      return o(h, d, g);
    if (y === Set)
      return p(h, d, g);
    var k = Er(h);
    return k === cr ? r(h, d, g) : k === hr ? c(h, d, g) : k === pr ? o(h, d, g) : k === gr ? p(h, d, g) : k === mr ? typeof h.then != "function" && typeof d.then != "function" && l(h, d, g) : k === yr ? f(h, d, g) : k === dr ? n(h, d, g) : k === lr ? l(h, d, g) : k === ur || k === fr || k === vr ? a(h, d, g) : !1;
  };
}
function wr(t) {
  var e = t.circular, r = t.createCustomConfig, n = t.strict, s = {
    areArraysEqual: n ? X : Jt,
    areDatesEqual: Qt,
    areErrorsEqual: er,
    areFunctionsEqual: tr,
    areMapsEqual: n ? Ze(Ve, X) : Ve,
    areNumbersEqual: rr,
    areObjectsEqual: n ? X : nr,
    arePrimitiveWrappersEqual: sr,
    areRegExpsEqual: or,
    areSetsEqual: n ? Ze(ze, X) : ze,
    areTypedArraysEqual: n ? X : ar,
    areUrlsEqual: ir
  };
  if (r && (s = He({}, s, r(s))), e) {
    var o = ie(s.areArraysEqual), i = ie(s.areMapsEqual), l = ie(s.areObjectsEqual), a = ie(s.areSetsEqual);
    s = He({}, s, {
      areArraysEqual: o,
      areMapsEqual: i,
      areObjectsEqual: l,
      areSetsEqual: a
    });
  }
  return s;
}
function Cr(t) {
  return function(e, r, n, s, o, i, l) {
    return t(e, r, l);
  };
}
function br(t) {
  var e = t.circular, r = t.comparator, n = t.createState, s = t.equals, o = t.strict;
  if (n)
    return function(a, c) {
      var p = n(), u = p.cache, f = u === void 0 ? e ? /* @__PURE__ */ new WeakMap() : void 0 : u, m = p.meta;
      return r(a, c, {
        cache: f,
        equals: s,
        meta: m,
        strict: o
      });
    };
  if (e)
    return function(a, c) {
      return r(a, c, {
        cache: /* @__PURE__ */ new WeakMap(),
        equals: s,
        meta: void 0,
        strict: o
      });
    };
  var i = {
    cache: void 0,
    equals: s,
    meta: void 0,
    strict: o
  };
  return function(a, c) {
    return r(a, c, i);
  };
}
var ce = N();
N({ strict: !0 });
N({ circular: !0 });
N({
  circular: !0,
  strict: !0
});
N({
  createInternalComparator: function() {
    return G;
  }
});
N({
  strict: !0,
  createInternalComparator: function() {
    return G;
  }
});
N({
  circular: !0,
  createInternalComparator: function() {
    return G;
  }
});
N({
  circular: !0,
  createInternalComparator: function() {
    return G;
  },
  strict: !0
});
function N(t) {
  t === void 0 && (t = {});
  var e = t.circular, r = e === void 0 ? !1 : e, n = t.createInternalComparator, s = t.createState, o = t.strict, i = o === void 0 ? !1 : o, l = wr(t), a = kr(l), c = n ? n(a) : Cr(a);
  return br({ circular: r, comparator: a, createState: s, equals: c, strict: i });
}
const Ke = [
  Int8Array,
  Uint8Array,
  Uint8ClampedArray,
  Int16Array,
  Uint16Array,
  Int32Array,
  Uint32Array,
  Float32Array,
  Float64Array
], me = 1, Y = 8;
class Se {
  /**
   * Creates an index from raw `ArrayBuffer` data.
   * @param {ArrayBuffer} data
   */
  static from(e) {
    if (!(e instanceof ArrayBuffer))
      throw new Error("Data must be an instance of ArrayBuffer.");
    const [r, n] = new Uint8Array(e, 0, 2);
    if (r !== 219)
      throw new Error("Data does not appear to be in a KDBush format.");
    const s = n >> 4;
    if (s !== me)
      throw new Error(`Got v${s} data when expected v${me}.`);
    const o = Ke[n & 15];
    if (!o)
      throw new Error("Unrecognized array type.");
    const [i] = new Uint16Array(e, 2, 1), [l] = new Uint32Array(e, 4, 1);
    return new Se(l, i, o, e);
  }
  /**
   * Creates an index that will hold a given number of items.
   * @param {number} numItems
   * @param {number} [nodeSize=64] Size of the KD-tree node (64 by default).
   * @param {TypedArrayConstructor} [ArrayType=Float64Array] The array type used for coordinates storage (`Float64Array` by default).
   * @param {ArrayBuffer} [data] (For internal use only)
   */
  constructor(e, r = 64, n = Float64Array, s) {
    if (isNaN(e) || e < 0)
      throw new Error(`Unpexpected numItems value: ${e}.`);
    this.numItems = +e, this.nodeSize = Math.min(Math.max(+r, 2), 65535), this.ArrayType = n, this.IndexArrayType = e < 65536 ? Uint16Array : Uint32Array;
    const o = Ke.indexOf(this.ArrayType), i = e * 2 * this.ArrayType.BYTES_PER_ELEMENT, l = e * this.IndexArrayType.BYTES_PER_ELEMENT, a = (8 - l % 8) % 8;
    if (o < 0)
      throw new Error(`Unexpected typed array class: ${n}.`);
    s && s instanceof ArrayBuffer ? (this.data = s, this.ids = new this.IndexArrayType(this.data, Y, e), this.coords = new this.ArrayType(this.data, Y + l + a, e * 2), this._pos = e * 2, this._finished = !0) : (this.data = new ArrayBuffer(Y + i + l + a), this.ids = new this.IndexArrayType(this.data, Y, e), this.coords = new this.ArrayType(this.data, Y + l + a, e * 2), this._pos = 0, this._finished = !1, new Uint8Array(this.data, 0, 2).set([219, (me << 4) + o]), new Uint16Array(this.data, 2, 1)[0] = r, new Uint32Array(this.data, 4, 1)[0] = e);
  }
  /**
   * Add a point to the index.
   * @param {number} x
   * @param {number} y
   * @returns {number} An incremental index associated with the added item (starting from `0`).
   */
  add(e, r) {
    const n = this._pos >> 1;
    return this.ids[n] = n, this.coords[this._pos++] = e, this.coords[this._pos++] = r, n;
  }
  /**
   * Perform indexing of the added points.
   */
  finish() {
    const e = this._pos >> 1;
    if (e !== this.numItems)
      throw new Error(`Added ${e} items when expected ${this.numItems}.`);
    return _e(this.ids, this.coords, this.nodeSize, 0, this.numItems - 1, 0), this._finished = !0, this;
  }
  /**
   * Search the index for items within a given bounding box.
   * @param {number} minX
   * @param {number} minY
   * @param {number} maxX
   * @param {number} maxY
   * @returns {number[]} An array of indices correponding to the found items.
   */
  range(e, r, n, s) {
    if (!this._finished)
      throw new Error("Data not yet indexed - call index.finish().");
    const { ids: o, coords: i, nodeSize: l } = this, a = [0, o.length - 1, 0], c = [];
    for (; a.length; ) {
      const p = a.pop() || 0, u = a.pop() || 0, f = a.pop() || 0;
      if (u - f <= l) {
        for (let g = f; g <= u; g++) {
          const v = i[2 * g], y = i[2 * g + 1];
          v >= e && v <= n && y >= r && y <= s && c.push(o[g]);
        }
        continue;
      }
      const m = f + u >> 1, h = i[2 * m], d = i[2 * m + 1];
      h >= e && h <= n && d >= r && d <= s && c.push(o[m]), (p === 0 ? e <= h : r <= d) && (a.push(f), a.push(m - 1), a.push(1 - p)), (p === 0 ? n >= h : s >= d) && (a.push(m + 1), a.push(u), a.push(1 - p));
    }
    return c;
  }
  /**
   * Search the index for items within a given radius.
   * @param {number} qx
   * @param {number} qy
   * @param {number} r Query radius.
   * @returns {number[]} An array of indices correponding to the found items.
   */
  within(e, r, n) {
    if (!this._finished)
      throw new Error("Data not yet indexed - call index.finish().");
    const { ids: s, coords: o, nodeSize: i } = this, l = [0, s.length - 1, 0], a = [], c = n * n;
    for (; l.length; ) {
      const p = l.pop() || 0, u = l.pop() || 0, f = l.pop() || 0;
      if (u - f <= i) {
        for (let g = f; g <= u; g++)
          Xe(o[2 * g], o[2 * g + 1], e, r) <= c && a.push(s[g]);
        continue;
      }
      const m = f + u >> 1, h = o[2 * m], d = o[2 * m + 1];
      Xe(h, d, e, r) <= c && a.push(s[m]), (p === 0 ? e - n <= h : r - n <= d) && (l.push(f), l.push(m - 1), l.push(1 - p)), (p === 0 ? e + n >= h : r + n >= d) && (l.push(m + 1), l.push(u), l.push(1 - p));
    }
    return a;
  }
}
function _e(t, e, r, n, s, o) {
  if (s - n <= r)
    return;
  const i = n + s >> 1;
  it(t, e, i, n, s, o), _e(t, e, r, n, i - 1, 1 - o), _e(t, e, r, i + 1, s, 1 - o);
}
function it(t, e, r, n, s, o) {
  for (; s > n; ) {
    if (s - n > 600) {
      const c = s - n + 1, p = r - n + 1, u = Math.log(c), f = 0.5 * Math.exp(2 * u / 3), m = 0.5 * Math.sqrt(u * f * (c - f) / c) * (p - c / 2 < 0 ? -1 : 1), h = Math.max(n, Math.floor(r - p * f / c + m)), d = Math.min(s, Math.floor(r + (c - p) * f / c + m));
      it(t, e, r, h, d, o);
    }
    const i = e[2 * r + o];
    let l = n, a = s;
    for (J(t, e, n, r), e[2 * s + o] > i && J(t, e, n, s); l < a; ) {
      for (J(t, e, l, a), l++, a--; e[2 * l + o] < i; )
        l++;
      for (; e[2 * a + o] > i; )
        a--;
    }
    e[2 * n + o] === i ? J(t, e, n, a) : (a++, J(t, e, a, s)), a <= r && (n = a + 1), r <= a && (s = a - 1);
  }
}
function J(t, e, r, n) {
  he(t, r, n), he(e, 2 * r, 2 * n), he(e, 2 * r + 1, 2 * n + 1);
}
function he(t, e, r) {
  const n = t[e];
  t[e] = t[r], t[r] = n;
}
function Xe(t, e, r, n) {
  const s = t - r, o = e - n;
  return s * s + o * o;
}
const Or = {
  minZoom: 0,
  // min zoom to generate clusters on
  maxZoom: 16,
  // max zoom level to cluster the points on
  minPoints: 2,
  // minimum points to form a cluster
  radius: 40,
  // cluster radius in pixels
  extent: 512,
  // tile extent (radius is calculated relative to it)
  nodeSize: 64,
  // size of the KD-tree leaf node, affects performance
  log: !1,
  // whether to log timing info
  // whether to generate numeric ids for input features (in vector tiles)
  generateId: !1,
  // a reduce function for calculating custom cluster properties
  reduce: null,
  // (accumulated, props) => { accumulated.sum += props.sum; }
  // properties to use for individual points when running the reducer
  map: (t) => t
  // props => ({sum: props.my_value})
}, Ye = Math.fround || ((t) => (e) => (t[0] = +e, t[0]))(new Float32Array(1)), $ = 2, j = 3, ge = 4, x = 5, lt = 6;
class ut {
  constructor(e) {
    this.options = Object.assign(Object.create(Or), e), this.trees = new Array(this.options.maxZoom + 1), this.stride = this.options.reduce ? 7 : 6, this.clusterProps = [];
  }
  load(e) {
    const { log: r, minZoom: n, maxZoom: s } = this.options;
    r && console.time("total time");
    const o = `prepare ${e.length} points`;
    r && console.time(o), this.points = e;
    const i = [];
    for (let a = 0; a < e.length; a++) {
      const c = e[a];
      if (!c.geometry)
        continue;
      const [p, u] = c.geometry.coordinates, f = Ye(le(p)), m = Ye(ue(u));
      i.push(
        f,
        m,
        // projected point coordinates
        1 / 0,
        // the last zoom the point was processed at
        a,
        // index of the source feature in the original input array
        -1,
        // parent cluster id
        1
        // number of points in a cluster
      ), this.options.reduce && i.push(0);
    }
    let l = this.trees[s + 1] = this._createTree(i);
    r && console.timeEnd(o);
    for (let a = s; a >= n; a--) {
      const c = +Date.now();
      l = this.trees[a] = this._createTree(this._cluster(l, a)), r && console.log("z%d: %d clusters in %dms", a, l.numItems, +Date.now() - c);
    }
    return r && console.timeEnd("total time"), this;
  }
  getClusters(e, r) {
    let n = ((e[0] + 180) % 360 + 360) % 360 - 180;
    const s = Math.max(-90, Math.min(90, e[1]));
    let o = e[2] === 180 ? 180 : ((e[2] + 180) % 360 + 360) % 360 - 180;
    const i = Math.max(-90, Math.min(90, e[3]));
    if (e[2] - e[0] >= 360)
      n = -180, o = 180;
    else if (n > o) {
      const u = this.getClusters([n, s, 180, i], r), f = this.getClusters([-180, s, o, i], r);
      return u.concat(f);
    }
    const l = this.trees[this._limitZoom(r)], a = l.range(le(n), ue(i), le(o), ue(s)), c = l.data, p = [];
    for (const u of a) {
      const f = this.stride * u;
      p.push(c[f + x] > 1 ? Je(c, f, this.clusterProps) : this.points[c[f + j]]);
    }
    return p;
  }
  getChildren(e) {
    const r = this._getOriginId(e), n = this._getOriginZoom(e), s = "No cluster with the specified id.", o = this.trees[n];
    if (!o)
      throw new Error(s);
    const i = o.data;
    if (r * this.stride >= i.length)
      throw new Error(s);
    const l = this.options.radius / (this.options.extent * Math.pow(2, n - 1)), a = i[r * this.stride], c = i[r * this.stride + 1], p = o.within(a, c, l), u = [];
    for (const f of p) {
      const m = f * this.stride;
      i[m + ge] === e && u.push(i[m + x] > 1 ? Je(i, m, this.clusterProps) : this.points[i[m + j]]);
    }
    if (u.length === 0)
      throw new Error(s);
    return u;
  }
  getLeaves(e, r, n) {
    r = r || 10, n = n || 0;
    const s = [];
    return this._appendLeaves(s, e, r, n, 0), s;
  }
  getTile(e, r, n) {
    const s = this.trees[this._limitZoom(e)], o = Math.pow(2, e), { extent: i, radius: l } = this.options, a = l / i, c = (n - a) / o, p = (n + 1 + a) / o, u = {
      features: []
    };
    return this._addTileFeatures(
      s.range((r - a) / o, c, (r + 1 + a) / o, p),
      s.data,
      r,
      n,
      o,
      u
    ), r === 0 && this._addTileFeatures(
      s.range(1 - a / o, c, 1, p),
      s.data,
      o,
      n,
      o,
      u
    ), r === o - 1 && this._addTileFeatures(
      s.range(0, c, a / o, p),
      s.data,
      -1,
      n,
      o,
      u
    ), u.features.length ? u : null;
  }
  getClusterExpansionZoom(e) {
    let r = this._getOriginZoom(e) - 1;
    for (; r <= this.options.maxZoom; ) {
      const n = this.getChildren(e);
      if (r++, n.length !== 1)
        break;
      e = n[0].properties.cluster_id;
    }
    return r;
  }
  _appendLeaves(e, r, n, s, o) {
    const i = this.getChildren(r);
    for (const l of i) {
      const a = l.properties;
      if (a && a.cluster ? o + a.point_count <= s ? o += a.point_count : o = this._appendLeaves(e, a.cluster_id, n, s, o) : o < s ? o++ : e.push(l), e.length === n)
        break;
    }
    return o;
  }
  _createTree(e) {
    const r = new Se(e.length / this.stride | 0, this.options.nodeSize, Float32Array);
    for (let n = 0; n < e.length; n += this.stride)
      r.add(e[n], e[n + 1]);
    return r.finish(), r.data = e, r;
  }
  _addTileFeatures(e, r, n, s, o, i) {
    for (const l of e) {
      const a = l * this.stride, c = r[a + x] > 1;
      let p, u, f;
      if (c)
        p = ct(r, a, this.clusterProps), u = r[a], f = r[a + 1];
      else {
        const d = this.points[r[a + j]];
        p = d.properties;
        const [g, v] = d.geometry.coordinates;
        u = le(g), f = ue(v);
      }
      const m = {
        type: 1,
        geometry: [[
          Math.round(this.options.extent * (u * o - n)),
          Math.round(this.options.extent * (f * o - s))
        ]],
        tags: p
      };
      let h;
      c || this.options.generateId ? h = r[a + j] : h = this.points[r[a + j]].id, h !== void 0 && (m.id = h), i.features.push(m);
    }
  }
  _limitZoom(e) {
    return Math.max(this.options.minZoom, Math.min(Math.floor(+e), this.options.maxZoom + 1));
  }
  _cluster(e, r) {
    const { radius: n, extent: s, reduce: o, minPoints: i } = this.options, l = n / (s * Math.pow(2, r)), a = e.data, c = [], p = this.stride;
    for (let u = 0; u < a.length; u += p) {
      if (a[u + $] <= r)
        continue;
      a[u + $] = r;
      const f = a[u], m = a[u + 1], h = e.within(a[u], a[u + 1], l), d = a[u + x];
      let g = d;
      for (const v of h) {
        const y = v * p;
        a[y + $] > r && (g += a[y + x]);
      }
      if (g > d && g >= i) {
        let v = f * d, y = m * d, k, P = -1;
        const O = ((u / p | 0) << 5) + (r + 1) + this.points.length;
        for (const E of h) {
          const w = E * p;
          if (a[w + $] <= r)
            continue;
          a[w + $] = r;
          const M = a[w + x];
          v += a[w] * M, y += a[w + 1] * M, a[w + ge] = O, o && (k || (k = this._map(a, u, !0), P = this.clusterProps.length, this.clusterProps.push(k)), o(k, this._map(a, w)));
        }
        a[u + ge] = O, c.push(v / g, y / g, 1 / 0, O, -1, g), o && c.push(P);
      } else {
        for (let v = 0; v < p; v++)
          c.push(a[u + v]);
        if (g > 1)
          for (const v of h) {
            const y = v * p;
            if (!(a[y + $] <= r)) {
              a[y + $] = r;
              for (let k = 0; k < p; k++)
                c.push(a[y + k]);
            }
          }
      }
    }
    return c;
  }
  // get index of the point from which the cluster originated
  _getOriginId(e) {
    return e - this.points.length >> 5;
  }
  // get zoom of the point from which the cluster originated
  _getOriginZoom(e) {
    return (e - this.points.length) % 32;
  }
  _map(e, r, n) {
    if (e[r + x] > 1) {
      const i = this.clusterProps[e[r + lt]];
      return n ? Object.assign({}, i) : i;
    }
    const s = this.points[e[r + j]].properties, o = this.options.map(s);
    return n && o === s ? Object.assign({}, o) : o;
  }
}
function Je(t, e, r) {
  return {
    type: "Feature",
    id: t[e + j],
    properties: ct(t, e, r),
    geometry: {
      type: "Point",
      coordinates: [Mr(t[e]), Sr(t[e + 1])]
    }
  };
}
function ct(t, e, r) {
  const n = t[e + x], s = n >= 1e4 ? `${Math.round(n / 1e3)}k` : n >= 1e3 ? `${Math.round(n / 100) / 10}k` : n, o = t[e + lt], i = o === -1 ? {} : Object.assign({}, r[o]);
  return Object.assign(i, {
    cluster: !0,
    cluster_id: t[e + j],
    point_count: n,
    point_count_abbreviated: s
  });
}
function le(t) {
  return t / 360 + 0.5;
}
function ue(t) {
  const e = Math.sin(t * Math.PI / 180), r = 0.5 - 0.25 * Math.log((1 + e) / (1 - e)) / Math.PI;
  return r < 0 ? 0 : r > 1 ? 1 : r;
}
function Mr(t) {
  return (t - 0.5) * 360;
}
function Sr(t) {
  const e = (180 - t * 360) * Math.PI / 180;
  return 360 * Math.atan(Math.exp(e)) / Math.PI - 90;
}
function Te(t, e) {
  var r = {};
  for (var n in t)
    Object.prototype.hasOwnProperty.call(t, n) && e.indexOf(n) < 0 && (r[n] = t[n]);
  if (t != null && typeof Object.getOwnPropertySymbols == "function")
    for (var s = 0, n = Object.getOwnPropertySymbols(t); s < n.length; s++)
      e.indexOf(n[s]) < 0 && Object.prototype.propertyIsEnumerable.call(t, n[s]) && (r[n[s]] = t[n[s]]);
  return r;
}
class C {
  static isAdvancedMarkerAvailable(e) {
    return google.maps.marker && e.getMapCapabilities().isAdvancedMarkersAvailable === !0;
  }
  static isAdvancedMarker(e) {
    return google.maps.marker && e instanceof google.maps.marker.AdvancedMarkerElement;
  }
  static setMap(e, r) {
    this.isAdvancedMarker(e) ? e.map = r : e.setMap(r);
  }
  static getPosition(e) {
    if (this.isAdvancedMarker(e)) {
      if (e.position) {
        if (e.position instanceof google.maps.LatLng)
          return e.position;
        if (Number.isFinite(e.position.lat) && Number.isFinite(e.position.lng))
          return new google.maps.LatLng(e.position.lat, e.position.lng);
      }
      return new google.maps.LatLng(null);
    }
    return e.getPosition();
  }
  static getVisible(e) {
    return this.isAdvancedMarker(e) ? !0 : e.getVisible();
  }
}
class te {
  constructor({ markers: e, position: r }) {
    this.markers = [], e && (this.markers = e), r && (r instanceof google.maps.LatLng ? this._position = r : this._position = new google.maps.LatLng(r));
  }
  get bounds() {
    if (this.markers.length === 0 && !this._position)
      return;
    const e = new google.maps.LatLngBounds(this._position, this._position);
    for (const r of this.markers)
      e.extend(C.getPosition(r));
    return e;
  }
  get position() {
    return this._position || this.bounds.getCenter();
  }
  /**
   * Get the count of **visible** markers.
   */
  get count() {
    return this.markers.filter((e) => C.getVisible(e)).length;
  }
  /**
   * Add a marker to the cluster.
   */
  push(e) {
    this.markers.push(e);
  }
  /**
   * Cleanup references and remove marker from map.
   */
  delete() {
    this.marker && (C.setMap(this.marker, null), this.marker = void 0), this.markers.length = 0;
  }
}
function q(t, e = "assertion failed") {
  if (t == null)
    throw Error(e);
}
const Tr = (t, e, r, n) => {
  const s = t.getBounds();
  q(s);
  const o = dt(s, e, n);
  return r.filter((i) => o.contains(C.getPosition(i)));
}, dt = (t, e, r) => {
  const { northEast: n, southWest: s } = Ar(t, e), o = qr({ northEast: n, southWest: s }, r);
  return Lr(o, e);
}, Pr = (t, e, r) => {
  const n = dt(t, e, r), s = n.getNorthEast(), o = n.getSouthWest();
  return [o.lng(), o.lat(), s.lng(), s.lat()];
}, Ar = (t, e) => {
  const r = e.fromLatLngToDivPixel(t.getNorthEast()), n = e.fromLatLngToDivPixel(t.getSouthWest());
  return q(r), q(n), { northEast: r, southWest: n };
}, qr = ({ northEast: t, southWest: e }, r) => (t.x += r, t.y -= r, e.x -= r, e.y += r, { northEast: t, southWest: e }), Lr = ({ northEast: t, southWest: e }, r) => {
  const n = r.fromDivPixelToLatLng(e), s = r.fromDivPixelToLatLng(t);
  return new google.maps.LatLngBounds(n, s);
};
class pt {
  constructor({ maxZoom: e = 16 }) {
    this.maxZoom = e;
  }
  /**
   * Helper function to bypass clustering based upon some map state such as
   * zoom, number of markers, etc.
   *
   * ```typescript
   *  cluster({markers, map}: AlgorithmInput): Cluster[] {
   *    if (shouldBypassClustering(map)) {
   *      return this.noop({markers})
   *    }
   * }
   * ```
   */
  noop({ markers: e }) {
    return jr(e);
  }
}
class xr extends pt {
  constructor(e) {
    var { viewportPadding: r = 60 } = e, n = Te(e, ["viewportPadding"]);
    super(n), this.viewportPadding = 60, this.viewportPadding = r;
  }
  calculate({ markers: e, map: r, mapCanvasProjection: n }) {
    const s = r.getZoom();
    return q(s), s >= this.maxZoom ? {
      clusters: this.noop({
        markers: e
      }),
      changed: !1
    } : {
      clusters: this.cluster({
        markers: Tr(r, n, e, this.viewportPadding),
        map: r,
        mapCanvasProjection: n
      })
    };
  }
}
const jr = (t) => t.map((r) => new te({
  position: C.getPosition(r),
  markers: [r]
}));
class Ir extends pt {
  constructor(e) {
    var { maxZoom: r, radius: n = 60 } = e, s = Te(e, ["maxZoom", "radius"]);
    super({ maxZoom: r }), this.markers = [], this.clusters = [], this.state = { zoom: -1 }, this.superCluster = new ut(Object.assign({ maxZoom: this.maxZoom, radius: n }, s));
  }
  calculate(e) {
    let r = !1, n = e.map.getZoom();
    q(n), n = Math.round(n);
    const s = { zoom: n };
    if (!ce(e.markers, this.markers)) {
      r = !0, this.markers = [...e.markers];
      const o = this.markers.map((i) => {
        const l = C.getPosition(i);
        return {
          type: "Feature",
          geometry: { type: "Point", coordinates: [l.lng(), l.lat()] },
          properties: { marker: i }
        };
      });
      this.superCluster.load(o);
    }
    return r || (this.state.zoom <= this.maxZoom || s.zoom <= this.maxZoom) && (r = !ce(this.state, s)), this.state = s, e.markers.length === 0 ? (this.clusters = [], { clusters: this.clusters, changed: r }) : (r && (this.clusters = this.cluster(e)), { clusters: this.clusters, changed: r });
  }
  cluster({ map: e }) {
    const r = e.getZoom();
    return q(r), this.superCluster.getClusters([-180, -90, 180, 90], Math.round(r)).map((n) => this.transformCluster(n));
  }
  transformCluster({ geometry: { coordinates: [e, r] }, properties: n }) {
    if (n.cluster)
      return new te({
        markers: this.superCluster.getLeaves(n.cluster_id, 1 / 0).map((o) => o.properties.marker),
        position: { lat: r, lng: e }
      });
    const s = n.marker;
    return new te({
      markers: [s],
      position: C.getPosition(s)
    });
  }
}
class Rr extends xr {
  constructor(e) {
    var { maxZoom: r, radius: n = 60, viewportPadding: s = 60 } = e, o = Te(e, ["maxZoom", "radius", "viewportPadding"]);
    super({ maxZoom: r, viewportPadding: s }), this.markers = [], this.clusters = [], this.superCluster = new ut(Object.assign({ maxZoom: this.maxZoom, radius: n }, o)), this.state = { zoom: -1, view: [0, 0, 0, 0] };
  }
  calculate(e) {
    const r = this.getViewportState(e);
    let n = !ce(this.state, r);
    if (!ce(e.markers, this.markers)) {
      n = !0, this.markers = [...e.markers];
      const s = this.markers.map((o) => {
        const i = C.getPosition(o);
        return {
          type: "Feature",
          geometry: {
            type: "Point",
            coordinates: [i.lng(), i.lat()]
          },
          properties: { marker: o }
        };
      });
      this.superCluster.load(s);
    }
    return n && (this.clusters = this.cluster(e), this.state = r), { clusters: this.clusters, changed: n };
  }
  cluster(e) {
    const r = this.getViewportState(e);
    return this.superCluster.getClusters(r.view, r.zoom).map((n) => this.transformCluster(n));
  }
  transformCluster({ geometry: { coordinates: [e, r] }, properties: n }) {
    if (n.cluster)
      return new te({
        markers: this.superCluster.getLeaves(n.cluster_id, 1 / 0).map((o) => o.properties.marker),
        position: { lat: r, lng: e }
      });
    const s = n.marker;
    return new te({
      markers: [s],
      position: C.getPosition(s)
    });
  }
  getViewportState(e) {
    const r = e.map.getZoom(), n = e.map.getBounds();
    return q(r), q(n), {
      zoom: Math.round(r),
      view: Pr(n, e.mapCanvasProjection, this.viewportPadding)
    };
  }
}
class Nr {
  constructor(e, r) {
    this.markers = { sum: e.length };
    const n = r.map((o) => o.count), s = n.reduce((o, i) => o + i, 0);
    this.clusters = {
      count: r.length,
      markers: {
        mean: s / r.length,
        sum: s,
        min: Math.min(...n),
        max: Math.max(...n)
      }
    };
  }
}
class $r {
  /**
   * The default render function for the library used by {@link MarkerClusterer}.
   *
   * Currently set to use the following:
   *
   * ```typescript
   * // change color if this cluster has more markers than the mean cluster
   * const color =
   *   count > Math.max(10, stats.clusters.markers.mean)
   *     ? "#ff0000"
   *     : "#0000ff";
   *
   * // create svg url with fill color
   * const svg = window.btoa(`
   * <svg fill="${color}" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 240 240">
   *   <circle cx="120" cy="120" opacity=".6" r="70" />
   *   <circle cx="120" cy="120" opacity=".3" r="90" />
   *   <circle cx="120" cy="120" opacity=".2" r="110" />
   *   <circle cx="120" cy="120" opacity=".1" r="130" />
   * </svg>`);
   *
   * // create marker using svg icon
   * return new google.maps.Marker({
   *   position,
   *   icon: {
   *     url: `data:image/svg+xml;base64,${svg}`,
   *     scaledSize: new google.maps.Size(45, 45),
   *   },
   *   label: {
   *     text: String(count),
   *     color: "rgba(255,255,255,0.9)",
   *     fontSize: "12px",
   *   },
   *   // adjust zIndex to be above other markers
   *   zIndex: 1000 + count,
   * });
   * ```
   */
  render({ count: e, position: r }, n, s) {
    const i = `<svg fill="${e > Math.max(10, n.clusters.markers.mean) ? "#ff0000" : "#0000ff"}" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 240 240" width="50" height="50">
<circle cx="120" cy="120" opacity=".6" r="70" />
<circle cx="120" cy="120" opacity=".3" r="90" />
<circle cx="120" cy="120" opacity=".2" r="110" />
<text x="50%" y="50%" style="fill:#fff" text-anchor="middle" font-size="50" dominant-baseline="middle" font-family="roboto,arial,sans-serif">${e}</text>
</svg>`, l = `Cluster of ${e} markers`, a = Number(google.maps.Marker.MAX_ZINDEX) + e;
    if (C.isAdvancedMarkerAvailable(s)) {
      const u = new DOMParser().parseFromString(i, "image/svg+xml").documentElement;
      u.setAttribute("transform", "translate(0 25)");
      const f = {
        map: s,
        position: r,
        zIndex: a,
        title: l,
        content: u
      };
      return new google.maps.marker.AdvancedMarkerElement(f);
    }
    const c = {
      position: r,
      zIndex: a,
      title: l,
      icon: {
        url: `data:image/svg+xml;base64,${btoa(i)}`,
        anchor: new google.maps.Point(25, 25)
      }
    };
    return new google.maps.Marker(c);
  }
}
function Br(t, e) {
  for (let r in e.prototype)
    t.prototype[r] = e.prototype[r];
}
class Pe {
  constructor() {
    Br(Pe, google.maps.OverlayView);
  }
}
var I;
(function(t) {
  t.CLUSTERING_BEGIN = "clusteringbegin", t.CLUSTERING_END = "clusteringend", t.CLUSTER_CLICK = "click", t.GMP_CLICK = "gmp-click";
})(I || (I = {}));
const Dr = (t, e, r) => {
  e.bounds && r.fitBounds(e.bounds);
};
class Zr extends Pe {
  constructor({ map: e, markers: r = [], algorithmOptions: n = {}, algorithm: s = new Ir(n), renderer: o = new $r(), onClusterClick: i = Dr }) {
    super(), this.map = null, this.idleListener = null, this.markers = [...r], this.clusters = [], this.algorithm = s, this.renderer = o, this.onClusterClick = i, e && this.setMap(e);
  }
  addMarker(e, r) {
    this.markers.includes(e) || (this.markers.push(e), r || this.render());
  }
  addMarkers(e, r) {
    e.forEach((n) => {
      this.addMarker(n, !0);
    }), r || this.render();
  }
  removeMarker(e, r) {
    const n = this.markers.indexOf(e);
    return n === -1 ? !1 : (C.setMap(e, null), this.markers.splice(n, 1), r || this.render(), !0);
  }
  removeMarkers(e, r) {
    let n = !1;
    return e.forEach((s) => {
      n = this.removeMarker(s, !0) || n;
    }), n && !r && this.render(), n;
  }
  clearMarkers(e) {
    this.markers.length = 0, e || this.render();
  }
  /**
   * Recalculates and draws all the marker clusters.
   */
  render() {
    const e = this.getMap();
    if (e instanceof google.maps.Map && e.getProjection()) {
      google.maps.event.trigger(this, I.CLUSTERING_BEGIN, this);
      const { clusters: r, changed: n } = this.algorithm.calculate({
        markers: this.markers,
        map: e,
        mapCanvasProjection: this.getProjection()
      });
      if (n || n == null) {
        const s = /* @__PURE__ */ new Set();
        for (const i of r)
          i.markers.length == 1 && s.add(i.markers[0]);
        const o = [];
        for (const i of this.clusters)
          i.marker != null && (i.markers.length == 1 ? s.has(i.marker) || C.setMap(i.marker, null) : o.push(i.marker));
        this.clusters = r, this.renderClusters(), requestAnimationFrame(() => o.forEach((i) => C.setMap(i, null)));
      }
      google.maps.event.trigger(this, I.CLUSTERING_END, this);
    }
  }
  onAdd() {
    const e = this.getMap();
    q(e), this.idleListener = e.addListener("idle", this.render.bind(this)), this.render();
  }
  onRemove() {
    this.idleListener && google.maps.event.removeListener(this.idleListener), this.reset();
  }
  reset() {
    this.markers.forEach((e) => C.setMap(e, null)), this.clusters.forEach((e) => e.delete()), this.clusters = [];
  }
  renderClusters() {
    const e = new Nr(this.markers, this.clusters), r = this.getMap();
    this.clusters.forEach((n) => {
      if (n.markers.length === 1)
        n.marker = n.markers[0];
      else if (n.marker = this.renderer.render(n, e, r), n.markers.forEach((s) => C.setMap(s, null)), this.onClusterClick) {
        const s = C.isAdvancedMarker(n.marker) ? I.GMP_CLICK : I.CLUSTER_CLICK;
        n.marker.addListener(
          s,
          /* istanbul ignore next */
          (o) => {
            google.maps.event.trigger(this, I.CLUSTER_CLICK, n), this.onClusterClick(o, n, r);
          }
        );
      }
      C.setMap(n.marker, r);
    });
  }
}
function Ee(t) {
  var e = typeof t;
  return t != null && (e == "object" || e == "function");
}
var Gr = typeof global == "object" && global && global.Object === Object && global;
const Ur = Gr;
var Fr = typeof self == "object" && self && self.Object === Object && self, Vr = Ur || Fr || Function("return this")();
const ft = Vr;
var zr = function() {
  return ft.Date.now();
};
const ve = zr;
var Wr = /\s/;
function Hr(t) {
  for (var e = t.length; e-- && Wr.test(t.charAt(e)); )
    ;
  return e;
}
var Kr = /^\s+/;
function Xr(t) {
  return t && t.slice(0, Hr(t) + 1).replace(Kr, "");
}
var Yr = ft.Symbol;
const de = Yr;
var mt = Object.prototype, Jr = mt.hasOwnProperty, Qr = mt.toString, Q = de ? de.toStringTag : void 0;
function en(t) {
  var e = Jr.call(t, Q), r = t[Q];
  try {
    t[Q] = void 0;
    var n = !0;
  } catch {
  }
  var s = Qr.call(t);
  return n && (e ? t[Q] = r : delete t[Q]), s;
}
var tn = Object.prototype, rn = tn.toString;
function nn(t) {
  return rn.call(t);
}
var sn = "[object Null]", on = "[object Undefined]", Qe = de ? de.toStringTag : void 0;
function an(t) {
  return t == null ? t === void 0 ? on : sn : Qe && Qe in Object(t) ? en(t) : nn(t);
}
function ln(t) {
  return t != null && typeof t == "object";
}
var un = "[object Symbol]";
function cn(t) {
  return typeof t == "symbol" || ln(t) && an(t) == un;
}
var et = 0 / 0, dn = /^[-+]0x[0-9a-f]+$/i, pn = /^0b[01]+$/i, fn = /^0o[0-7]+$/i, mn = parseInt;
function tt(t) {
  if (typeof t == "number")
    return t;
  if (cn(t))
    return et;
  if (Ee(t)) {
    var e = typeof t.valueOf == "function" ? t.valueOf() : t;
    t = Ee(e) ? e + "" : e;
  }
  if (typeof t != "string")
    return t === 0 ? t : +t;
  t = Xr(t);
  var r = pn.test(t);
  return r || fn.test(t) ? mn(t.slice(2), r ? 2 : 8) : dn.test(t) ? et : +t;
}
var hn = "Expected a function", gn = Math.max, vn = Math.min;
function yn(t, e, r) {
  var n, s, o, i, l, a, c = 0, p = !1, u = !1, f = !0;
  if (typeof t != "function")
    throw new TypeError(hn);
  e = tt(e) || 0, Ee(r) && (p = !!r.leading, u = "maxWait" in r, o = u ? gn(tt(r.maxWait) || 0, e) : o, f = "trailing" in r ? !!r.trailing : f);
  function m(E) {
    var w = n, M = s;
    return n = s = void 0, c = E, i = t.apply(M, w), i;
  }
  function h(E) {
    return c = E, l = setTimeout(v, e), p ? m(E) : i;
  }
  function d(E) {
    var w = E - a, M = E - c, L = e - w;
    return u ? vn(L, o - M) : L;
  }
  function g(E) {
    var w = E - a, M = E - c;
    return a === void 0 || w >= e || w < 0 || u && M >= o;
  }
  function v() {
    var E = ve();
    if (g(E))
      return y(E);
    l = setTimeout(v, d(E));
  }
  function y(E) {
    return l = void 0, f && n ? m(E) : (n = s = void 0, i);
  }
  function k() {
    l !== void 0 && clearTimeout(l), c = 0, n = a = s = l = void 0;
  }
  function P() {
    return l === void 0 ? i : y(ve());
  }
  function O() {
    var E = ve(), w = g(E);
    if (n = arguments, s = this, a = E, w) {
      if (l === void 0)
        return h(a);
      if (u)
        return clearTimeout(l), l = setTimeout(v, e), m(a);
    }
    return l === void 0 && (l = setTimeout(v, e)), i;
  }
  return O.cancel = k, O.flush = P, O;
}
class _n extends Zr {
  constructor(r, n = 10) {
    super(r);
    ae(this, "debouncedRender");
    this.debouncedRender = yn(
      () => {
        super.render();
      },
      n,
      { leading: !0, trailing: !0 }
    );
  }
  addMarker(r, n) {
    super.addMarker(r, !0), n || this.debouncedRender();
  }
  removeMarker(r, n) {
    const s = super.removeMarker(r, !0);
    return n || this.debouncedRender(), s;
  }
  addMarkers(r, n) {
    super.addMarkers(r, !0), n || this.debouncedRender();
  }
  removeMarkers(r, n) {
    const s = super.removeMarkers(r, !0);
    return n || this.debouncedRender(), s;
  }
  clearMarkers(r) {
    super.clearMarkers(!0), r || this.debouncedRender();
  }
  /** Renders immediately, canceling any pending debounced render. */
  render() {
    this.debouncedRender.cancel(), super.render();
  }
  destroy() {
    this.debouncedRender.cancel();
  }
}
class En extends Rr {
  calculate(e) {
    return e.markers.length === 0 ? { clusters: [], changed: !0 } : super.calculate(e);
  }
}
const rt = Object.values(I), Rn = T({
  name: "MarkerCluster",
  props: {
    options: {
      type: Object,
      default: () => ({})
    },
    renderDebounceDelay: {
      type: Number,
      default: 10
    }
  },
  emits: rt,
  setup(t, { emit: e, expose: r, slots: n }) {
    const s = _(), o = b(D, _()), i = b(Z, _());
    return U(Oe, s), S(
      o,
      () => {
        o.value && (s.value = A(
          new _n(
            {
              map: o.value,
              // Better perf than the default `SuperClusterAlgorithm`. See:
              // https://github.com/googlemaps/js-markerclusterer/pull/640
              algorithm: new En(t.options.algorithmOptions ?? {}),
              ...t.options
            },
            t.renderDebounceDelay
          )
        ), rt.forEach((l) => {
          var a;
          (a = s.value) == null || a.addListener(l, (c) => e(l, c));
        }));
      },
      {
        immediate: !0
      }
    ), B(() => {
      var l;
      s.value && ((l = i.value) == null || l.event.clearInstanceListeners(s.value), s.value.clearMarkers(!0), s.value.setMap(null), s.value.destroy());
    }), r({ markerCluster: s }), () => {
      var l;
      return (l = n.default) == null ? void 0 : l.call(n);
    };
  }
}), kn = T({
  inheritAttrs: !1,
  props: {
    options: {
      type: Object,
      required: !0
    }
  },
  setup(t, { slots: e, emit: r, expose: n }) {
    const s = _(), o = W(() => {
      var a;
      return (a = e.default) == null ? void 0 : a.call(e).some((c) => c.type !== ke);
    }), i = W(() => ({
      ...t.options,
      element: s.value
    })), l = K(fe, [], i, r);
    return n({
      customMarker: l
    }), { customMarkerRef: s, customMarker: l, hasSlotContent: o };
  }
});
const wn = {
  key: 0,
  class: "custom-marker-wrapper"
};
function Cn(t, e, r, n, s, o) {
  return t.hasSlotContent ? (F(), V("div", wn, [
    pe("div", we({
      ref: "customMarkerRef",
      style: { cursor: t.$attrs.onClick ? "pointer" : void 0 }
    }, t.$attrs), [
      z(t.$slots, "default", {}, void 0, !0)
    ], 16)
  ])) : Ce("", !0);
}
const Nn = /* @__PURE__ */ re(kn, [["render", Cn], ["__scopeId", "data-v-574a274a"]]), $n = T({
  name: "HeatmapLayer",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(t) {
    const e = _(), r = b(D, _()), n = b(Z, _());
    return S(
      [r, () => t.options],
      ([s, o], [i, l]) => {
        const a = !ee(o, l) || r.value !== i;
        if (r.value && n.value && a) {
          let c;
          if (o.data && !(o.data instanceof n.value.MVCArray)) {
            const p = n.value.LatLng, u = o.data.map((f) => f instanceof p || "location" in f && (f.location instanceof p || f.location === null) ? f : "location" in f ? { ...f, location: new p(f.location) } : new p(f));
            c = {
              ...o,
              data: u
            };
          } else
            c = o;
          e.value ? e.value.setOptions(c) : e.value = A(
            new n.value.visualization.HeatmapLayer({
              ...c,
              map: r.value
            })
          );
        }
      },
      { immediate: !0 }
    ), B(() => {
      e.value && e.value.setMap(null);
    }), { heatmapLayer: e };
  },
  render: () => null
});
export {
  Tn as AdvancedMarker,
  xn as Circle,
  jn as CustomControl,
  Nn as CustomMarker,
  Sn as GoogleMap,
  $n as HeatmapLayer,
  In as InfoWindow,
  Pn as Marker,
  Rn as MarkerCluster,
  qn as Polygon,
  An as Polyline,
  Ln as Rectangle,
  Z as apiSymbol,
  D as mapSymbol,
  st as mapTilesLoadedSymbol
};
